src/nbl/video/IOpenGL_PhysicalDeviceBase.h

```diff
@@ -28,7 +28,7 @@ class IOpenGLPhysicalDeviceBase : public IPhysicalDevice
 
 		IOpenGLPhysicalDeviceBase(
 			core::smart_refctd_ptr<system::ISystem>&& s,
-			core::smart_refctd_ptr<asset::IGLSLCompiler>&& glslc,
+			core::smart_refctd_ptr<asset::CGLSLCompiler>&& glslc,
```

Comment:
not yet, but in my plan ;)
We need an `m_api` in IPhysicalDevice so anytime we need the compiler: logicalDevice -> physicalDevice -> apiConnection -> smart_refctd<CCompilerSet> 
and if that's too many indirections, we can just hold the refctd inside the logical device.

Reply:
APIConnection holds physical devices, that's what I meant

Reply:
I just found out I refactored all this shit in my other branch. that's one of the reasons I had to finalize that first, I moved the compile to api connection because it was getting created for every physicalDevice.
https://github.com/Devsh-Graphics-Programming/Nabla/blob/erfan_image_upload_utils/src/nbl/video/IOpenGL_PhysicalDeviceBase.h#L29
After I merge that I want to make a few changes:
- Even remove the compiler from apiconnection (doesn't make much sense to me)
- The user will create a compilerSet and gives a `smart_refctd<CCompilerSet>` that to ILogicalDevice::SCreationParams, If it was nullptr, LogicalDevice will create one (we've discussed this before that logical device takes a compiler set to enable sharing)
- PhysicalDevice and APIConnection then have nothing to do with compilers and I will regain my sanity